### DISCUSSION TITLE
[Feature] Add Support for DIM Modules

### DISCUSSION THREAD

**Discussion by St0rml** [TillFleisch/ESPHome-PHC-Controller#4] [General]
Hey,
first off, thanks for the great project! I plan to integrate the existing installation of PEHA in my parent's house to homeassistant. However, most of the light modules in the house are DIM modules.
Now, I've spent some time with the code and tried to copy the structure of existing code for JRM and AMD modules in my code. I also checked the documentation for the PEHA BUS protocol as well as the esphome docs, however, I barely am a hobbyist programmer and have little to no knowledge of C++. So, this all is a little hard to grasp for me, especially with these bitwise operations.
Therefore, I would kindly like to ask if you could give me some help on the implementation and help with your experience and expertise.
I don't want to make a PR yet, as I really can't tell, if my code is somewhat correct, and also I've had no opportunity to test it out in an actual deployment.
I hope this is the right channel for this kind of help/question...
Find my fork here: https://github.com/St0rml/ESPHome-PHC-Controller

**Comment by TillFleisch** [comment]
I has been a while since I have had a look at this project. Unfortunately, I don't have access to a DIM module and I'm in a situation similar to yours, the PHC system is at my parent's place.
I'm currently very time constraint so I can't promise help any time soon :/.

I would recommend, to first get part of the system running too see if this project fits your needs before investing more time. Pick a couple of supportd modules and disconnect the rest and use a small ESPHome/YAML config to get started.

As for the DIM module, you could test your implementation in a simulation/test environment. You could use a simple python script which imitates the DIM module/commands (according to the documentation).
Given a usb to serial converter, you can test the ESPs functionality this way, without requiring a PHC system.
Furthermore, I would recommend having a look at different ESPHome light components, to see how they are implemented (regarding the dimming functionality and transition times). 

I will later converter this issue into a discussion, once I'm not on mobile.
